app/Exceptions/Handler.php

```diff
@@ -60,17 +85,23 @@ protected function unauthenticated($request, AuthenticationException $exception)
             return response()->json(['error' => 'Unauthenticated.'], 401);
         }
         // For later if we have different users.
+
         $guard = array_get($exception->guards(), 0);
         switch ($guard) {
             case 'api':
                 $login = route('api.login');
                 break;
+            //TODO: merge
+            case 'store':
+                $login = route('store.login');
+                break;
             case 'admin':
             default:
                 $login = route('admin.login');
                 break;
         }
 
-        return redirect()->guest($login);
+        return redirect()->guest($login)
+            ->withErrors(['error_message' => trans('store.auth.exception')]);
```

Comment:
Do admin and api also need error messages? or is store a left over namespace and the message will be same for all users?

Reply:
Admin and api will ignore the messagebag object.

Reply:
OK